crypto/x509/by_store.c

```diff
@@ -115,27 +135,54 @@ static int by_store_ctrl_ex(X509_LOOKUP *ctx, int cmd, const char *argp,
     switch (cmd) {
     case X509_L_ADD_STORE:
         if (argp != NULL) {
-            STACK_OF(OPENSSL_STRING) *uris = X509_LOOKUP_get_method_data(ctx);
-            char *data = OPENSSL_strdup(argp);
+            STACK_OF(CACHED_STORE) *stores = X509_LOOKUP_get_method_data(ctx);
+            CACHED_STORE *data = OPENSSL_malloc(sizeof(*data));
 
             if (data == NULL) {
                 return 0;
             }
-            if (uris == NULL) {
-                uris = sk_OPENSSL_STRING_new_null();
-                X509_LOOKUP_set_method_data(ctx, uris);
+            memset(data, 0, sizeof(*data));
+            data->uri = OPENSSL_strdup(argp);
+            data->libctx = libctx;
+            if (propq != NULL)
+                data->propq = OPENSSL_strdup(propq);
```

Comment:
Why dup'ing the string here, which is not done in `cache_objects()`?
After removing that, zero'ing the struct fields (e.g., via `memset()`) will no more be needed, as well as the below check on `data->propq`.

Reply:
Ah, I haven't really finished...  `by_store_subject_ex()` really shouldn't be a thing at all with these changes.  It's designed to override the libctx and propq that was passed through `by_store_ctrl_ex()`, which just seems *very* *wrong*.

Reply:
But, coming back to the question of duplication.  I duplicate here for exactly the same reason that we already duplicate the URI.  I hope this is reasonable at this point.